StardewArchipelago/Locations/CodeInjections/Vanilla/PhoneInjections.cs

```diff
@@ -65,5 +97,439 @@ public static bool CallAdventureGuild_AllowRecovery_Prefix(DefaultPhoneHandler _
+                        Game1.DrawDialogue(characterFromName, "Strings\\Characters:Phone_Clint_Working", value);
+                    }
+                }
+                else
+                {
+                    string scheduleKey = characterFromName.ScheduleKey;
+                    if (!(scheduleKey == "winter_16"))
+                    {
+                        if (scheduleKey == "Fri")
+                        {
+                            Game1.DrawAnsweringMachineDialogue(characterFromName, "Strings\\Characters:Phone_Clint_Festival");
+                        }
+                        else if (Game1.timeOfDay >= 900 && Game1.timeOfDay < 1600)
+                        {
+                            Game1.DrawDialogue(characterFromName, "Strings\\Characters:Phone_Clint_Open" + ((Game1.random.NextDouble() < 0.01) ? "_Rare" : ""));
+                        }
+                        else
+                        {
+                            Game1.DrawAnsweringMachineDialogue(characterFromName, "Strings\\Characters:Phone_Clint_Closed");
+                        }
+                    }
+                    else
+                    {
+                        Game1.DrawAnsweringMachineDialogue(characterFromName, "Strings\\Characters:Phone_Clint_Festival");
+                    }
+                }
+
+                Game1.afterDialogues = (Game1.afterFadeFunction)Delegate.Combine(Game1.afterDialogues, (Game1.afterFadeFunction)delegate
+                {
+                    Response[] answerChoices = new Response[3]
+                    {
+                        new Response("BlacksmithEntrance", "Check Entrance"),
+                        new Response("Blacksmith_UpgradeCost", Game1.content.LoadString("Strings\\Characters:Phone_CheckToolCost")),
+                        new Response("HangUp", Game1.content.LoadString("Strings\\Characters:Phone_HangUp"))
+                    };
+                    location.createQuestionDialogue(Game1.content.LoadString("Strings\\Characters:Phone_SelectOption"), answerChoices, "telephone");
+                });
+            }, 4950);
+            return false;
+        }
+
+        public static bool AnswerDialogueAction_BlacksmithEntrance_Prefix(GameLocation __instance, string questionAndAnswer, string[] questionParams)
+        {
+            if (questionAndAnswer != "telephone_BlacksmithEntrance")
+            {
+                return true;
+            }
+
+            var character = Game1.getCharacterFromName("Clint");
+            if (GameLocation.AreStoresClosedForFestival() || character.ScheduleKey == "Fri" || character.ScheduleKey == "winter_16" || Game1.timeOfDay <= 900 || Game1.timeOfDay > 1600)
+            {
+                Dialogue locationDialog = new Dialogue(character, "", $"I am currently located at the {_entranceManager.GetCurrentModifiedEntranceTo("Blacksmith")} today. Come in through the {_entranceManager.GetCurrentModifiedEntranceFrom("Blacksmith")}. *Click*");

```

Comment:
There is a significant amount of duplication everywhere in this code, for the dialogue parts and all.
You should
1: Centralize the behavior (A EntrancePhoneHandler class?)
2: Extract relevant string constants, and reuse the ones that can be reused
3: These methods have so much nesting, you need to extract methods and simplify branches a lot more than that